File: idt/src/infrastructure/persistence/models/chunking_profile.py
"""ChunkingProfile SQLAlchemy 모델 (clause-aware-chunking Design §4.1)."""
from datetime import datetime

from sqlalchemy import JSON, DateTime, Index, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from src.infrastructure.persistence.models.base import Base


class ChunkingProfileModel(Base):
    __tablename__ = "chunking_profile"
    __table_args__ = (
        Index("idx_chunking_profile_status", "status"),
        Index("idx_chunking_profile_default", "is_default", "status"),
    )

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    description: Mapped[str | None] = mapped_column(String(500), nullable=True)
    boundary_rules: Mapped[list] = mapped_column(JSON, nullable=False)
    parent_chunk_size: Mapped[int] = mapped_column(
        Integer, nullable=False, default=2000
    )
    chunk_size: Mapped[int] = mapped_column(Integer, nullable=False, default=500)
    chunk_overlap: Mapped[int] = mapped_column(Integer, nullable=False, default=50)
    is_default: Mapped[bool] = mapped_column(Integer, nullable=False, default=0)
    # card-section-summary D2: 섹션 요약 LLM 소프트 참조 (FK 없음, NULL=비활성)
    summary_llm_model_id: Mapped[str | None] = mapped_column(
        String(36), nullable=True
    )
    status: Mapped[str] = mapped_column(
        String(20), nullable=False, default="active"
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=func.now(), onupdate=func.now()
    )

File: idt_front/src/types/chunkingProfile.ts
// chunking-profile-admin-ui: 관리자 청킹 프로파일 타입
// 계약 원본: idt/src/api/routes/admin_chunking_router.py

export type BoundaryRuleLevel = 'parent' | 'child';

export interface BoundaryRule {
  pattern: string;
  priority: number;
  level: BoundaryRuleLevel;
}

export interface ChunkingProfile {
  profile_id: string;
  name: string;
  description: string | null;
  boundary_rules: BoundaryRule[];
  parent_chunk_size: number;
  chunk_size: number;
  chunk_overlap: number;
  is_default: boolean;
  // 섹션 요약 LLM (card-section-summary D2) — null이면 요약 비활성
  summary_llm_model_id: string | null;
  created_at: string | null;
  updated_at: string | null;
}

/** 생성(POST)/수정(PUT) 공용 바디 — PUT은 전체 교체이므로 모든 필드 필수 전송 */
export interface ChunkingProfileRequest {
  name: string;
  description: string | null;
  boundary_rules: BoundaryRule[];
  parent_chunk_size: number;
  chunk_size: number;
  chunk_overlap: number;
  is_default: boolean;
  summary_llm_model_id: string | null;
}

export interface ChunkingProfileListResponse {
  profiles: ChunkingProfile[];
  total: number;
}

export interface ChunkingProfileMessageResponse {
  profile_id: string;
  message: string;
}
